modules/fedimint-ln-client/src/pay.rs

```diff
@@ -26,42 +27,53 @@ use crate::{set_payment_result, LightningClientContext, LightningClientStateMach
 
 const RETRY_DELAY: Duration = Duration::from_secs(1);
 
-#[cfg_attr(doc, aquamarine::aquamarine)]
-/// State machine that requests the lightning gateway to pay an invoice on
-/// behalf of a federation client.
-///
-/// ```mermaid
-/// graph LR
-/// classDef virtual fill:#fff,stroke-dasharray: 5 5
-///
-///  CreatedOutgoingLnContract -- await transaction failed --> Canceled
-///  CreatedOutgoingLnContract -- await transaction acceptance --> Funded
-///  Funded -- await gateway payment success  --> Success
-///  Funded -- await gateway payment failed --> Refundable
-///  Refundable -- gateway issued refunded --> Refund
-///  Refundable -- transaction timeout --> Refund
-///  Refund -- await transaction acceptance --> Refunded
-///  Refund -- await transaction rejected --> Failure
-/// ```
-#[allow(clippy::large_enum_variant)]
-#[derive(Debug, Clone, Eq, PartialEq, Hash, Decodable, Encodable)]
-pub enum LightningPayStates {
-    CreatedOutgoingLnContract(LightningPayCreatedOutgoingLnContract),
-    FundingRejected,
-    Funded(LightningPayFunded),
-    Success(String),
-    #[deprecated(
-        since = "0.4.0",
-        note = "Pay State Machine skips over this state and will retry payments until cancellation or timeout"
-    )]
-    Refundable(LightningPayRefundable),
-    Refund(LightningPayRefund),
-    #[deprecated(
-        since = "0.4.0",
-        note = "Pay State Machine does not need to wait for the refund tx to be accepted"
-    )]
-    Refunded(Vec<OutPoint>),
-    Failure(String),
+#[allow(deprecated)]
+pub mod lightningpay {
```

Comment:
`pub(crate)` or `pub(super)`.

Reply:
And a comment why we need this module.

Reply:
done

Reply:
ah unfortunately needs to be `pub` since it is used in the testing crate